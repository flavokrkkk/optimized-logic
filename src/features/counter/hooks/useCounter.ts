import { CounterContext } from "@/entities";
import { useContext } from "react";
export const useCounter = () => {
  const context = useContext(CounterContext);

  if (!context) {
    throw new Error("useCounter must be used within a CounterProvider");
  }

  return context;
};
