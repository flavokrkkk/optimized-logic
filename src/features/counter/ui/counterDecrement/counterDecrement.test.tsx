import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { CounterProvider } from "@/entities";
import CounterDecrement from "../counterDecrement/counterDecrement";
import { useCounter } from "../../hooks/useCounter";

jest.mock("../../hooks/useCounter", () => ({
  useCounter: jest.fn(),
}));

describe("CounterDecrement", () => {
  it("should render the decrement button correctly", () => {
    const mockHandleDecrement = jest.fn();
    (useCounter as jest.Mock).mockReturnValue({
      handleDecrement: mockHandleDecrement,
    });

    render(
      <CounterProvider>
        <CounterDecrement />
      </CounterProvider>
    );

    expect(screen.getByText("-")).toBeInTheDocument();
  });

  it("should call handleDecrement when button is clicked", () => {
    const mockHandleDecrement = jest.fn();
    (useCounter as jest.Mock).mockReturnValue({
      handleDecrement: mockHandleDecrement,
    });

    render(
      <CounterProvider>
        <CounterDecrement />
      </CounterProvider>
    );

    const button = screen.getByText("-");
    fireEvent.click(button);

    expect(mockHandleDecrement).toHaveBeenCalledTimes(1);
  });

  it("should match snapshot", () => {
    const mockHandleDecrement = jest.fn();
    (useCounter as jest.Mock).mockReturnValue({
      handleDecrement: mockHandleDecrement,
    });

    const { asFragment } = render(
      <CounterProvider>
        <CounterDecrement />
      </CounterProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
