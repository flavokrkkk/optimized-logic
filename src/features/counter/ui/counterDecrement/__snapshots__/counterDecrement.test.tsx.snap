// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterDecrement should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn primary medium enabled"
  >
    <span
      class="content-group"
    >
      -
    </span>
  </button>
</DocumentFragment>
`;
